src/main/java/redis/clients/jedis/MasterListener.java

```diff
@@ -0,0 +1,110 @@
+package redis.clients.jedis;
+
+import redis.clients.jedis.errors.SilentErrorHandler;
+import redis.clients.jedis.exceptions.JedisConnectionException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+* Created by piotrturek on 29/03/15.
+*/
+class MasterListener extends Thread {
+
+    private ShardedJedisSentinelPool shardedJedisSentinelPool;
+    protected List<String> masters;
+    protected String host;
+    protected int port;
+    protected Jedis jedis;
+    protected AtomicBoolean running = new AtomicBoolean(false);
+    private MasterListenerErrorHandler errorHandler;
+
+    protected MasterListener(ShardedJedisSentinelPool shardedJedisSentinelPool) {
+        this.shardedJedisSentinelPool = shardedJedisSentinelPool;
+    }
+
+    public MasterListener(ShardedJedisSentinelPool shardedJedisSentinelPool, List<String> masters, String host, int port) {
+        this.shardedJedisSentinelPool = shardedJedisSentinelPool;
+        this.masters = masters;
+        this.host = host;
+        this.port = port;
+        this.errorHandler = new SilentErrorHandler("Sentinel at " + host + " and port " + port, shardedJedisSentinelPool.log);
```

Comment:
There are two reasons:
- keep the behaviour consistent with previous version of the library and log as ConnectionPool
- make it testable